docs/src/main/asciidoc/websockets-next-reference.adoc

```diff
@@ -783,6 +787,67 @@ class MyBean {
 [[websocket-next-security]]
 === Security
 
+Security capabilities are provided by the Quarkus Security extension.
+Any xref:security-identity-providers.adoc[Identity provider] can be used to convert authentication credentials on the initial HTTP request into a `SecurityIdentity` instance.
+The `SecurityIdentity` is then associated with the websocket connection.
+Authorization options are demonstrated in following sections.
+
+NOTE: When an OpenID Connect extension, `quarkus-oidc`, is used and token expires, Quarkus automatically closes connection.
+
+[[secure-http-upgrade]]
+==== Secure HTTP upgrade
+
+An HTTP upgrade is secured when a standard security annotation is placed on an endpoint class or an HTTP Security policy is defined.
+The advantage of securing HTTP upgrade is less processing, the authorization is performed early and only once.
```

Comment:
IMHO lines 801 and 802 may need to be reworked. IMHO it is really a question whether users want to have a secure upgrade or not as opposed to optimizing the security checks (or securing callback methods only), since the HTTP upgrade is the main entry point... 

May be we can just drop lines 801, 802 ? Line 800 is good

Reply:
I don't mind dropping 801, but I think it is important for users to understand the advantage of securing HTTP upgrade. The advantage is that you perform security check once instead of on every message, which in case of WS can be  a lot. Maybe you can suggest rephrasing 802, please?

Reply:
@michalvavrik Sure, let us just move on with this PR and consider tuning the text which was already there before later